### ISSUE TITLE
"Failed to parse unit files" error on Updates page with Cockpit 243

### ISSUE THREAD

**Issue by AdamWill** [cockpit-project/cockpit#15764]
Cockpit version: 243
OS: Fedora Linux
Page: Updates

With Cockpit 243 on Fedora Linux 33 and 34, the Updates page shows an error "Failed to parse unit files for dnf-automatic.timer or dnf-automatic-installer.timer. Please remove custom overrides to enable this feature." and the button is labeled "Enable" rather than "Edit" (I have not yet tested if clicking it actually lets you configure automatic updates or not). This worked fine in 233 (it showed no error, and the button was labelled "Edit").

![cockpitautoerror](https://user-images.githubusercontent.com/916551/116623363-fc1d4880-a8fa-11eb-8e8d-dd6fba94f64f.png)

**Comment by AdamWill** [comment]
This is on a clean, fresh test system automatically deployed by openQA. It is in vanilla state and does not have any local automatic update configuration.

**Comment by AdamWill** [comment]
If I set openQA to just go ahead and click Enable anyway, it works. But the message is confusing and reads like it *shouldn't* work.

**Comment by martinpitt** [comment]
Thanks for your report @AdamWill !

See https://openqa.fedoraproject.org/tests/871851#step/server_cockpit_autoupdate/9

I just manually tested this on our Fedora 34 server cloud image with dnf-automatic-4.6.1-1.fc34.noarch.  There is nothing dnf-ish in `systemctl --all -t timer` or `systemctl list-timers`. So we need to find out the output of

    systemctl cat dnf-automatic{,-install}.timer

from that openQA system. Is that possible somehow?

We can certainly trigger this with some known-unsupported (on our page) timer formats, like "on every February day at 11:00":
```sh
mkdir -p /etc/systemd/system/dnf-automatic-install.timer.d
printf "[Timer]\nOnUnitInactiveSec=\nOnCalendar=*-02-* 11:00\n" > /etc/systemd/system/dnf-automatic-install.timer.d/time.conf
systemctl daemon-reload
systemctl start dnf-automatic-install.timer
```

Then with 243 one can still edit/override that. But indeed we could argue that we better not stomp over that customization, and disable the edit button and the status ("Not set up"). I think we should do that, but of course that won't help the openQA tests -- it will just make them "fail harder".

**Comment by AdamWill** [comment]
A thought occurs: did you test what it does if dnf-automatic is not installed at all? Because that's how it starts out in the openQA test. cockpit has the ability to install it during the configuration phase, and does so when we go ahead, ignoring the warning. So the warning should not be shown in that case.

**Comment by martinpitt** [comment]
@AdamWill : Hah, score! (And now I feel ashamed of not thinking of that..)

**Comment by martinpitt** [comment]
Sent a fix to #15767
